velox/expression/Expr.cpp

```diff
@@ -481,8 +481,9 @@ class ExprExceptionContext {
  public:
   ExprExceptionContext(
       const Expr* FOLLY_NONNULL expr,
-      const RowVector* FOLLY_NONNULL vector)
-      : expr_(expr), vector_(vector) {}
+      const RowVector* FOLLY_NONNULL vector,
+      const ExprSet* FOLLY_NULLABLE parentExprSet)
```

Comment:
nit: Instead of using raw pointer/FOLLY_NULLABLE etc if we want to denote optionality why not just use optional ?

Reply:
Unfortunately, there is no good way of passing around pointers when using std::optional which is why i had to default back to passing raw pointers. here is a good discussion around this point: https://www.reddit.com/r/cpp_questions/comments/m0sgm7/best_practices_regarding_stdoptional/